packages/backend/src/apps/google-sheets/actions/find-worksheet/index.ts

```diff
@@ -0,0 +1,187 @@
+  arguments: [
+    {
+      label: 'Drive',
+      key: 'driveId',
+      type: 'dropdown' as const,
+      required: false,
+      description:
+        'The Google Drive where your spreadsheet resides. If nothing is selected, then your personal Google Drive will be used.',
+      variables: true,
+      source: {
+        type: 'query',
+        name: 'getDynamicData',
+        arguments: [
+          {
+            name: 'key',
+            value: 'listDrives',
+          },
+        ],
+      },
+    },
+    {
+      label: 'Spreadsheet',
+      key: 'spreadsheetId',
+      type: 'dropdown' as const,
+      required: true,
+      dependsOn: ['parameters.driveId'],
+      variables: true,
+      source: {
+        type: 'query',
+        name: 'getDynamicData',
+        arguments: [
+          {
+            name: 'key',
+            value: 'listSpreadsheets',
+          },
+          {
+            name: 'parameters.driveId',
+            value: '{parameters.driveId}',
+          },
+        ],
+      },
+    },
+    {
+      label: 'Title',
+      key: 'title',
+      type: 'string' as const,
+      required: true,
+      description:
+        'The worksheet title needs to match exactly, and the search is case-sensitive.',
+      variables: true,
+    },
+    {
+      label: 'Create worksheet if none are found.',
+      key: 'createWorksheet',
+      type: 'dropdown' as const,
+      required: false,
+      options: [
+        { label: 'Yes', value: true },
+        { label: 'No', value: false },
+      ],
+      additionalFields: {
+        type: 'query',
+        name: 'getDynamicFields',
+        arguments: [
+          {
+            name: 'key',
+            value: 'listCreateWorksheetFields',
+          },
+          {
+            name: 'parameters.createWorksheet',
+            value: '{parameters.createWorksheet}',
+          },
+        ],
+      },
+    },
+  ],
+
+  async run($) {
+    const createWorksheet = $.step.parameters.createWorksheet;
+
+    const {
+      data: { sheets },
+    } = await $.http.get<TSheetsResponse>(
+      `/v4/spreadsheets/${$.step.parameters.spreadsheetId}`
+    );
+    const selectedSheet = sheets.find(
+      (sheet) => sheet.properties.title === $.step.parameters.title
+    );
+
+    if (createWorksheet) {
+      const headers = $.step.parameters.headers as THeaders;
+      const headerValues = headers.map((entry) => entry.header);
+      const overwrite = $.step.parameters.overwrite;
+
+      const body: TBody = {
+        requests: [
+          {
+            addSheet: {
+              properties: {
+                title: $.step.parameters.title,
+              },
+            },
+          },
+        ],
+      };
+
+      if (overwrite === 'true' && selectedSheet) {
+        body.requests.unshift({
+          deleteSheet: {
+            sheetId: selectedSheet.properties.sheetId,
+          },
+        });
+      }
+
+      const { data } = await $.http.post(
```

Comment:
This logic shouldn't throw an error in the following scenario;

- Set a sample worksheet title (that actually exists)
- True to "create a worksheet if none are found"
- Provide a sample header value
- No to "Overwrite"

In this case, we should respond with the existing worksheet itself rather than exposing an error just because we decided not to overwrite but failed to create it. We shouldn't have tried to create it in the place.